sms/src/main/java/org/motechproject/sms/configs/Config.java

```diff
@@ -152,17 +152,17 @@ public void setProps(List<ConfigProp> props) {
     }
 
     /**
-     * Returns an appropriate {@link DeliveryStatus} for the given failure count. If the failure count is higher than
-     * the maximum number of retries for this configuration, then {@link DeliveryStatus#ABORTED} is returned. Otherwise
-     * {@link DeliveryStatus#RETRYING} is returned, indicating the retries are ongoing.
+     * Returns an appropriate Delivery Status for the given failure count. If the failure count is higher than
+     * the maximum number of retries for this configuration, then ABORTED Status is returned. Otherwise
+     * RETRYING Status is returned, indicating the retries are ongoing.
      * @param failureCount the failure count to check
      * @return the delivery status matching the failure count
      */
-    public DeliveryStatus retryOrAbortStatus(Integer failureCount) {
+    public String retryOrAbortStatus(Integer failureCount) {
```

Comment:
Those methods are checking the number of failed attempts to sent the sms and returning the proper status (retry or abort) in order to save sms record with this status in database and resend sms when status is retry. The provider response does not have proper failure status and we still need retry mechanims so I don't see purpose in removing those methods